internal/pkg/vm/actor/storage.go

```diff
@@ -85,6 +85,23 @@ func ReadState(ctx runtime.Runtime, st interface{}) error {
 	return nil
 }
 
+// WriteState stores state and commits it as the actor's head
+func WriteState(ctx runtime.Runtime, state interface{}) error {
+	stage := ctx.Storage()
+
+	cid, err := stage.Put(state)
+	if err != nil {
+		return vmerrors.RevertErrorWrap(err, "Could not stage memory chunk")
```

Comment:
I know this is just taking from `WithState` but I think these should be fault errors.  They all trace back to ipld cbor and blockstore errors.